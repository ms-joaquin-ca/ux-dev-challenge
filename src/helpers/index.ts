export * from './styles'
export * from './tests'
