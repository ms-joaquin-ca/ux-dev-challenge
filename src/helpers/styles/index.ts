export { default as Normalize } from './normalize'
