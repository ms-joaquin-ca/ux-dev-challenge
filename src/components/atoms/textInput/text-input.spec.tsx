import { render } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import type { TextInputProps } from 'typings/components'

import TextInput from './text-input'
import { provideTheme } from 'helpers'

const defaultProps: TextInputProps = {
  label: 'Test Input',
  value: 'Test Value',
  id: 'test-input-id',
}

const onChange = jest.fn()

const setup = (props: TextInputProps) =>
  render(provideTheme(<TextInput onChange={onChange} {...props} />))

describe('TextInput', () => {
  it('should render a default text-input', () => {
    const wrapper = setup(defaultProps)

    expect(wrapper.container).toMatchSnapshot()
  }),
    it('should change the value of the input', () => {
      const wrapper = setup(defaultProps)

      const input = wrapper.getByLabelText('Test Input')

      userEvent.type(input, 'Test')

      expect(onChange).toHaveBeenCalledTimes(4)
    })
})
