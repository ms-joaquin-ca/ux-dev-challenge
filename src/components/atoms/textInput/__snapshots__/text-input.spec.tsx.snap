// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput should render a default text-input 1`] = `
<div>
  <div
    class="sc-bcXHqe gcKSdl"
  >
    <label
      for="test-input-id"
    >
      Test Input
    </label>
    <input
      id="test-input-id"
      type="text"
      value="Test Value"
    />
    <span />
  </div>
</div>
`;
